homeassistant/components/ping/device_tracker.py

```diff
@@ -118,15 +118,18 @@ async def async_update(now):
 
         async def async_update(now):
             """Update all the hosts on every interval time."""
-            responses = await hass.async_add_executor_job(
-                partial(
-                    multiping,
-                    ip_to_dev_id.keys(),
-                    count=PING_ATTEMPTS_COUNT,
-                    timeout=ICMP_TIMEOUT,
-                    privileged=privileged,
-                    id=async_get_next_ping_id(hass, len(ip_to_dev_id)),
-                )
+            responses = await gather_with_concurrency(
```

Comment:
The `async_multiping` function looks a lot like what you want to do 😉

Reply:
Thanks. Adjusted